com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ActivitiExceptionEventHandler.java

```diff
@@ -28,34 +30,44 @@ public ActivitiExceptionEventHandler(ProgressMessageService progressMessageServi
         this.progressMessageService = progressMessageService;
     }
 
-    public void handle(ActivitiEvent event) {
-        if (!(event instanceof ActivitiExceptionEvent)) {
+    public void handle(FlowableEvent event) {
+        if (!(event instanceof FlowableExceptionEvent) && !(event instanceof FlowableEngineEvent)) {

```

Comment:
Why check for both types when you're casting only to FlowableExceptionEvent?

Reply:
Because the below method is using the FlowableEngineEvent

Reply:
But why do both checks? I assume that FlowableEngineEvent extends FlowableExceptionEvent (or vice versa). Otherwise, one of the casts is bound to fail.